Sources/PostgresKit/PostgresConnectionSource.swift

```diff
@@ -9,36 +12,59 @@ public struct PostgresConnectionSource: ConnectionPoolSource {
         logger: Logger,
         on eventLoop: EventLoop
     ) -> EventLoopFuture<PostgresConnection> {
-        let address: SocketAddress
-        do {
-            address = try self.configuration.address()
-        } catch {
-            return eventLoop.makeFailedFuture(error)
-        }
-        return PostgresConnection.connect(
-            to: address,
-            tlsConfiguration: self.configuration.tlsConfiguration,
-            serverHostname: self.configuration._hostname,
-            logger: logger,
-            on: eventLoop
-        ).flatMap { conn in
-            return conn.authenticate(
-                username: self.configuration.username,
-                database: self.configuration.database,
-                password: self.configuration.password,
+        if let hostname = self.configuration._hostname {
+            let future = PostgresConnection.connect(
+                on: eventLoop,
+                configuration: .init(
+                    connection: .init(host: hostname, port: self.configuration._port ?? PostgresConfiguration.ianaPortNumber),
+                    authentication: .init(username: self.configuration.username, database: self.configuration.database, password: self.configuration.password),
+                    tls: self.configuration.tlsConfiguration.map { .require(try! .init(configuration: $0)) } ?? .disable
```

Comment:
Please generate the NIOSSLContext once only. Creating a context is a very, very expensive operation and should only be done once per pool.